src/comm/aca_grpc.cpp

```diff
@@ -102,4 +112,12 @@ void GoalStateProvisionerImpl::RunServer()
   ACA_LOG_INFO("Streaming capable GRPC server listening on %s\n",
                GRPC_SERVER_ADDRESS.c_str());
   server->Wait();
+  ACA_LOG_INFO("%s\n", "Trying to init a new sub to connect to the NCM");
+  // grpc::ChannelArguments args;
+  // args.SetChannelArgs();
+  // grpc::CreateCustomChannel(NCM_ADDRESS + ":" + NCM_PORT,
+  //                           grpc::InsecureChannelCredentials(), );
+  stub_ = GoalStateProvisioner::NewStub(grpc::CreateChannel(
```

Comment:
we need to move the grpc client code before the grpc server code because "server->Wait();" is designed to stuck there. 

we will want to rename this function from RunServer to RunClientAndServer :)